store/postgres/src/store.rs

```diff
@@ -209,59 +328,74 @@ impl Store {
         Ok(())
     }
 
-    /// Handles entity changes emitted by Postgres.
-    fn handle_entity_changes(
-        &self,
-        entity_changes: Box<Stream<Item = EntityChange, Error = ()> + Send>,
-    ) {
+    /// Handles change events emitted by Postgres.
+    fn handle_store_events(&self, store_events: Box<Stream<Item = StoreEvent, Error = ()> + Send>) {
         let logger = self.logger.clone();
         let subscriptions = self.subscriptions.clone();
+        let dirty_blocks = self.dirty_blocks.clone();
 
-        tokio::spawn(entity_changes.for_each(move |change| {
-            trace!(logger, "Received entity change event";
+        tokio::spawn(store_events.for_each(move |event| {
+            match &event {
+                StoreEvent::Change(change) => {
+                    trace!(logger, "Received entity change event";
                            "subgraph_id" => change.subgraph_id.to_string(),
                            "entity_type" => &change.entity_type,
                            "entity_id" => &change.entity_id);
+                }
+                StoreEvent::Tick(tick) => {
+                    trace!(logger, "Received block tick"; "block" => tick.block.to_string());
+                }
+            }
+
+            // Record which BlockTick subscriptions have had 'interesting'
+            // changes and are therefore dirty
+            for (id, subscription) in subscriptions.read().unwrap().iter() {
+                if subscription.is_dirty(&event) {
```

Comment:
What confuses me a little here is that non-BlockTick subscriptions are also added to subscriptions with changes yet. So the block tick concept isn't separate. Another reason why is rename `is_dirty` and `DirtyBlocks`.